client/sniffer/src/main/java/org/elasticsearch/client/sniff/ElasticsearchHostsSniffer.java

```diff
@@ -89,68 +92,112 @@ public ElasticsearchHostsSniffer(RestClient restClient, long sniffRequestTimeout
     /**
      * Calls the elasticsearch nodes info api, parses the response and returns all the found http hosts
      */
-    public List<HttpHost> sniffHosts() throws IOException {
+    @Override
+    public List<Node> sniffHosts() throws IOException {
         Response response = restClient.performRequest("get", "/_nodes/http", sniffRequestParams);
         return readHosts(response.getEntity());
     }
 
-    private List<HttpHost> readHosts(HttpEntity entity) throws IOException {
+    private List<Node> readHosts(HttpEntity entity) throws IOException {
         try (InputStream inputStream = entity.getContent()) {
             JsonParser parser = jsonFactory.createParser(inputStream);
             if (parser.nextToken() != JsonToken.START_OBJECT) {
                 throw new IOException("expected data to start with an object");
             }
-            List<HttpHost> hosts = new ArrayList<>();
+            List<Node> nodes = new ArrayList<>();
             while (parser.nextToken() != JsonToken.END_OBJECT) {
                 if (parser.getCurrentToken() == JsonToken.START_OBJECT) {
                     if ("nodes".equals(parser.getCurrentName())) {
                         while (parser.nextToken() != JsonToken.END_OBJECT) {
                             JsonToken token = parser.nextToken();
                             assert token == JsonToken.START_OBJECT;
                             String nodeId = parser.getCurrentName();
-                            HttpHost sniffedHost = readHost(nodeId, parser, this.scheme);
-                            if (sniffedHost != null) {
-                                logger.trace("adding node [" + nodeId + "]");
-                                hosts.add(sniffedHost);
-                            }
+                            readHost(nodeId, parser, scheme, nodes);
```

Comment:
do we need to pass in the whole list of nodes? can't we still return a node and add it like before?

Reply:
Probably! let me fix it.